test/trento/vault_test.exs

```diff
@@ -0,0 +1,77 @@
+defmodule Trento.VaultTest do
+  @moduledoc false
+
+  use ExUnit.Case
+
+  alias Trento.Vault
+
+  @moduletag :integration
+
+  describe "Encryption" do
+    test "should encrypt/decrypt plain content" do
+      secrets = ["secret1", "secret2", "secret3"]
+
+      Enum.each(secrets, fn secret ->
+        encrypted = Vault.encrypt!(secret)
+        refute secret == encrypted
+
+        decrypted = Vault.decrypt!(encrypted)
+        assert secret == decrypted
+      end)
+    end
+  end
+
+  describe "Decryption" do
+    test "should decrypt to plain content" do
+      encrypted_secrets = [
+        secret1: "AQpBRVMuR0NNLlYxQKbSB9raisFu6xsWwmT3ql0R0pdfpMffV9fyzvOBg4KT/vc=",
+        secret2: "AQpBRVMuR0NNLlYxg02iIWC4RH1NMOc7QlzSA3fcwUQ243IaFP+qGbJMQmQgT44=",
+        secret3: "AQpBRVMuR0NNLlYxhAx+UMg0XkqCyAXPv7UlNg7xVeig4sMLrwapysrGNvD8A9k="
+      ]
+
+      Enum.each(encrypted_secrets, fn {decrypted, encrypted} ->
+        plaintext = encrypted |> Base.decode64!() |> Vault.decrypt!()
+        assert plaintext == Atom.to_string(decrypted)
+      end)
+    end
+
+    defmodule RuntimeVault do
+      @moduledoc """
+      A vault used for testing purposes
+      """
+      use Cloak.Vault, otp_app: :trento
+    end
+
+    test "should not decrypt when the content has been encrypted with a different key" do
```

Comment:
No need to test it, Cloak is already tested, and I'm not sure anyway this is the right way to inject a genserver into a test